packages/network/src/workers/syncUtils.ts

```diff
@@ -350,7 +353,10 @@ export async function fetchStateInBlockRange(
     setPercentage
   );
 
-  storeEvents(cacheStore, events);
+  storeEvents(
+    cacheStore,
+    filter(events, (e) => isNetworkComponentUpdateEvent(e)) as NetworkComponentUpdate<Components>[]
```

Comment:
why `lodash` instead of the built in `Array.filter` (which also supports type guards)?

```suggestion
    events.filter(isNetworkComponentUpdateEvent)
```